frontend/src/app/patients/Components/PersonForm.test.tsx

```diff
@@ -0,0 +1,138 @@
+    employedInHealthcare: undefined,
+    ethnicity: "refused",
+    facilityId: null,
+    firstName: "",
+    gender: "refused",
+    lastName: "",
+    lookupId: "",
+    middleName: "",
+    phoneNumbers: null,
+    preferredLanguage: null,
+    race: "refused",
+    residentCongregateSetting: undefined,
+    role: "",
+    state: "",
+    street: "",
+    streetTwo: null,
+    telephone: "",
+    testResultDelivery: undefined,
+    tribalAffiliation: undefined,
+    zipCode: "",
+  };
+  const savePersonFunction = (
+    _savePerson: Nullable<PersonFormData>,
+    _startTest?: boolean
+  ) => {};
+  const getFooter = (
+    _onSave: (startTest?: boolean) => void,
+    _formChanged: boolean
+  ) => <div></div>;
+  const mockStore = createMockStore([]);
+  const navigationContext = {
+    basename: "",
+    navigator: {
+      block: jest.fn().mockImplementation((fn) => {
+        fn();
+      }),
+      push: jest.fn() as any,
+      replace: jest.fn() as any,
+      go: jest.fn() as any,
+      createHref: jest.fn() as any,
+    },
+    static: true,
+  };
+  const store = {
+    facilities: [
+      {
+        id: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
+        name: "Testing Site",
+      },
+    ],
+  };
+  describe("date of birth", () => {
+    describe("validation", () => {
+      beforeEach(() => {
+        render(
+          <Provider store={mockStore({ ...store })}>
+            <UnsafeNavigationContext.Provider value={navigationContext}>
+              <PersonForm
+                patient={personFormData}
+                savePerson={savePersonFunction}
+                getFooter={getFooter}
+              />
+            </UnsafeNavigationContext.Provider>
+          </Provider>
+        );
+      });
+      it("should have an error message after invalid date was inserted ", async () => {
+        let element = screen.getByTestId("date-picker-external-input");
+        userEvent.type(element, "07/04/1776");
+        fireEvent.blur(element);
+        await waitFor(() => {
+          expect(screen.getByText("Error:")).toBeInTheDocument();
+        });
+      });
+      it("should remove the error after a valid date was inserted", async () => {
+        let element = screen.getByTestId("date-picker-external-input");
+        userEvent.type(element, "07/04/1776");
+        fireEvent.blur(element);
+        await waitFor(() => {
+          expect(screen.getByText("Error:")).toBeInTheDocument();
+        });
+        userEvent.clear(element);
+        userEvent.type(element, "07/04/2022");
+        fireEvent.blur(element);
+
+        await waitFor(() => {
+          expect(screen.queryByText("Error:")).not.toBeInTheDocument();
+        });
+      });
+      describe("pxpView", () => {
+        it("should have not have error message after blur", async () => {
+          cleanup();
+          render(
+            <Provider store={mockStore({ ...store })}>
+              <UnsafeNavigationContext.Provider value={navigationContext}>
```

Comment:
Can you explain the usage of this UnsafeNavigationContext @BobanL? I haven't seen it before and I'm not quite sure why it's necessary here.

Reply:
So it's actually used in [Prompt](https://github.com/CDCgov/prime-simplereport/blob/boban/4075-add-patient-date-picker/frontend/src/app/utils/Prompt.tsx#L6) which gets used in [PersonForm](https://github.com/CDCgov/prime-simplereport/blob/boban/4075-add-patient-date-picker/frontend/src/app/patients/Components/PersonForm.tsx#L361), so after a bunch of searching and testing I decided just using the provider to set the context would work (which I guess it does). 
But in this case, since I'm not even testing or needing to mock that navigation context it's definitely not a great idea to do that if not needed. So instead we can just use MemoryRouter which will provide that needed navigation context to prompt 😅.

Reply:
Ahh I see, thank you for the explanation! 😄